index.js

```diff
@@ -181,6 +174,8 @@ Pjax.prototype = {
       else if (request.getResponseHeader("X-XHR-Redirected-To")) {
         href = request.getResponseHeader("X-XHR-Redirected-To")
       }
+      this.state.href = href
+      this.state.options = options
```

Comment:
I would avoid mutation if possible and prefer a new object here.

Reply:
Mutation of what?

Reply:
this.state. Imo it's better to avoid a mutation and create a new state (in case you want to compare a state change, it's easier to do state === prevState)
I am mostly doing react this days, and I was thinking about this https://reactjs.org/docs/react-component.html#shouldcomponentupdate :)

Reply:
I'm still not sure what you mean. `state` is just a name I picked for the property to store the current `href` and `options`, so we can get it after the switches call back. After that, it gets destroyed ([here](https://github.com/MoOx/pjax/pull/110/files#diff-168726dbe96b3ce427e7fedce31bb0bcR263)). We could have used `this.currentHref` for the same results.

We could pass them around instead of making it a global property, but that gets messy and confusing.

Reply:
Unless you mean we should do

    this.state = {
        href: href,
        options: options
    }
instead of

    this.state.href = href
    this.state.options = options

Reply:
That's what I mean. But you will probably need to copy the other state key/value.
Anyway, if you feel it's totally useless, feel free to move forward and ignore my comment. I am just quickly reviewing here as I currently don't use this library on any active project.

Reply:
I don't see any reason to need the state once we're done with it, or to keep track of previous states, so I'm going to leave it for now.